lib/devicetrust/native/device_windows.go

```diff
@@ -0,0 +1,489 @@
+	home, err := getHomeDir()
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	deviceStateDirPath := path.Join(home, deviceStateFolderName)
+	keyPath := path.Join(deviceStateDirPath, attestationKeyFileName)
+
+	_, err = os.Stat(deviceStateDirPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			// If it doesn't exist, we can create it and return as we know
+			// the perms are correct as we created it.
+			if err := os.Mkdir(deviceStateDirPath, 700); err != nil {
+				return "", trace.Wrap(err)
+			}
+			return keyPath, nil
+		}
+		return "", trace.Wrap(err)
+	}
+
+	return keyPath, nil
+}
+
+func openTPM() (*attest.TPM, error) {
+	cfg := &attest.OpenConfig{
+		TPMVersion: attest.TPMVersion20,
+	}
+
+	tpm, err := attest.OpenTPM(cfg)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return tpm, nil
+}
+
+// getMarshaledEK returns the EK public key in PKIX, ASN.1 DER format.
+func getMarshaledEK(tpm *attest.TPM) ([]byte, error) {
+	eks, err := tpm.EKs()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if len(eks) == 0 {
+		return nil, trace.BadParameter("no endorsement keys found in tpm")
+	}
+	// TODO(noah): Marshal EK Certificate instead of key if present:
+	// https://github.com/gravitational/teleport.e/issues/1393
+	encodedEK, err := x509.MarshalPKIXPublicKey(eks[0].Public)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return encodedEK, nil
+}
+
+// loadOrCreateAK attempts to load an AK from disk. A NotFound error will be
+// returned if no such file exists.
+func loadAK(
+	tpm *attest.TPM,
+	persistencePath string,
+) (*attest.AK, error) {
+	ref, err := os.ReadFile(persistencePath)
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+
+	ak, err := tpm.LoadAK(ref)
+	if err != nil {
+		return nil, trace.Wrap(err, "loading ak into tpm")
+	}
+
+	return ak, nil
+}
+
+func createAndSaveAK(
+	tpm *attest.TPM,
+	persistencePath string,
+) (*attest.AK, error) {
+	ak, err := tpm.NewAK(nil)
+	if err != nil {
+		return nil, trace.Wrap(err, "creating ak")
+	}
+
+	// Write it to the well-known location on disk
+	ref, err := ak.Marshal()
+	if err != nil {
+		return nil, trace.Wrap(err, "marshalling ak")
+	}
+	err = os.WriteFile(persistencePath, ref, 600)
+	if err != nil {
+		return nil, trace.Wrap(err, "writing ak to disk")
+	}
+
+	return ak, nil
+}
+
+func enrollDeviceInit() (*devicepb.EnrollDeviceInit, error) {
+	akPath, err := setupDeviceStateDir(os.UserHomeDir)
+	if err != nil {
+		return nil, trace.Wrap(err, "setting up device state directory")
+	}
+
+	tpm, err := openTPM()
+	if err != nil {
+		return nil, trace.Wrap(err, "opening tpm")
+	}
+	defer tpm.Close()
+
+	// Try to load an existing AK in the case of re-enrollment, but, if the
+	// AK does not exist, create one and persist it.
+	ak, err := loadAK(tpm, akPath)
+	if err != nil {
+		if !trace.IsNotFound(err) {
+			return nil, trace.Wrap(err, "loading ak")
+		}
+		log.Debug("No existing AK was found on disk, an AK will be created.")
```

Comment:
Suggestion: prefix logs with TPM? Makes for easier grepping.

```suggestion
		log.Debug("TPM: No existing AK was found on disk, an AK will be created.")
```

Same where applicable.

Reply:
Yeah - this seems like a good idea.

Reply:
972eaa933e